python/lsst/afw/image/testUtils.py

```diff
@@ -36,125 +41,108 @@ def makeGaussianNoiseMaskedImage(dimensions, sigma, variance=1.0):
+
+    @throw AssertionError if any of the following are true for unmasked pixels:
+    - non-finite values differ in any way (e.g. one pixel is "nan" and another is not)
+    - finite values differ by too much
     """
-    retStr = ""
-    if skipMaskArr is not None:
-        skipMaskArr = numpy.array(skipMaskArr != 0, dtype=bool)
-        maskedArr1 = numpy.ma.array(maskArr1, copy=False, mask = skipMaskArr)
-        maskedArr2 = numpy.ma.array(maskArr2, copy=False, mask = skipMaskArr)
-        filledArr1 = maskedArr1.filled(0.0)
-        filledArr2 = maskedArr2.filled(0.0)
-    else:
-        filledArr1 = maskArr1
-        filledArr2 = maskArr2
+    errStr = imagesDiffer(mask0, mask1, skipMask=skipMask, rtol=0, atol=0)
+    if errStr:
+        testCase.fail("%s: %s" % (msg, errStr))
 
-    if numpy.any(filledArr1 != filledArr2):
-        errArr = numpy.abs(filledArr1 - filledArr2)
-        maxErr = errArr.max()
-        maxPosInd = numpy.where(errArr==maxErr)
-        maxPosTuple = (maxPosInd[1][0], maxPosInd[0][0])
-        retStr = "maxDiff=%s at position %s; value=%s vs. %s" % \
-            (maxErr, maxPosTuple, filledArr1[maxPosInd][0], filledArr2[maxPosInd][0])
-        retStr = "masks differ"
-    return retStr
-
-def maskedImagesDiffer(maskedImageArrSet1, maskedImageArrSet2,
-    doImage=True, doMask=True, doVariance=True, skipMaskArr=None, rtol=1.0e-05, atol=1e-08):
-    """Compare pixels from two masked images
-    
-    Inputs:
-    - maskedImageArrSet1: first masked image to compare as (image, mask, variance) arrays
-    - maskedImageArrSet2: second masked image to compare as (image, mask, variance) arrays
-    - doImage: compare image planes if True
-    - doMask: compare mask planes if True
-    - doVariance: compare variance planes if True
-    - skipMaskArr: pixels to ingore on the image, mask and variance arrays; nonzero values are skipped
-    - rtol: relative tolerance (see below)
-    - atol: absolute tolerance (see below)
+@lsst.utils.tests.inTestCase
+def assertMaskedImagesNearlyEqual(testCase, maskedImage0, maskedImage1,
+    doImage=True, doMask=True, doVariance=True, skipMask=None,
+    rtol=1.0e-05, atol=1e-08, msg="Masked images differ"):
+    """!Assert that two masked images are nearly equal, including non-finite values
     
-    rtol and atol are positive, typically very small numbers.
-    The relative difference (rtol * abs(b)) and the absolute difference "atol" are added together
-    to compare against the absolute difference between "a" and "b".
+    @param[in] testCase  unittest.TestCase instance the test is part of;
+                        an object supporting one method: fail(self, msgStr)
+    @param[in] maskedImage0  masked image 0 (an lsst.afw.image.MaskedImage or
+        collection of three transposed numpy arrays: image, mask, variance)
+    @param[in] maskedImage1  masked image 1 (an lsst.afw.image.MaskedImage or
+        collection of three transposed numpy arrays: image, mask, variance)
```

Comment:
Yes, and if the two images and the mask are all un-transposed and the same size then it will work. In the special case of 3 numpy arrays we only care that they are the same shape.